### PR TITLE
Allow specified short versions of options

### LINKED ISSUE #1: Similar flags are overriding each other
Hi,

I noticed this bug while working with your library where if you have flags that start similarly then they'll override each other.

```
assert 'OptParser#opts with similar flags' do
  parser = OptParser.new
  parser.on(:"export-targets", :string, "linux,windows")
  parser.on(:"export-directory", :string, "./exports")
  parser.parse(['--export-directory', './releases'])

  assert_equal({
    "export-targets": "linux,windows",
    "export-directory": "./releases",
  }, parser.opts)
end
```

### PR DESCRIPTION
**Pull request by HellRok**
This resolves #1

### CODE REVIEW

**Comment by HellRok** [comment]
Just noticed a bug with this, please don't merge until I've updated

**Comment by HellRok** [comment]
False alarm, but I did add some more tests to make sure

File: mrblib/opt_parser.rb [hunk 1]
```diff
@@ -40,19 +40,23 @@ def initialize
...
   # @param [ Symbol ] type The type of the option v
-  # @param [ Object ] dval The value to use if nothing else given.
+  #                        Possible values: object, string, int, float, bool
+  # @param [ Object ] default The value to use if nothing else given.
+  # @param [ String ] short The short name of the option value
   # @param [ Proc ]   blk  The callback to be invoked.
   #
   # @return [ Void ]
-  def on(opt, type = :object, dval = nil, &blk)
+  def on(opt, type = :object, default: nil, short: nil, &blk)
```
Comment (katzer) [inline mrblib/opt_parser.rb]: I dont like the breaking change to be honest. The syntax is propably better and more flexible but not backward compatible.

I think thats better:

```ruby
def on(opt, type = :object, dval = nil, short: nil, &blk)
```

**Review by katzer — CHANGES_REQUESTED** [review]
Many thanks btw for the PR. I'm sorry that I just saw it now.

**Comment by HellRok** [comment]
Happy to help out, I've made the requested changes. Please let me know if there's any other changes you want

**Comment by katzer** [comment]
@HellRok Now you just renamed the kwarg. What I mean is that the arg for default value should continue to be a simple argument like as it is.

**Comment by HellRok** [comment]
Oops, that's my bad. I've now put it back to a standard arg @katzer

File: README.md [hunk 2]
```diff
@@ -4,13 +4,14 @@ Tiny option parser for [mruby][mruby] with __zero__ dependencies.
 
 ```ruby
 parser = OptParser.new do |opts|
-  opts.on(:port, :int, 80) { |port| ... }
-  opts.on(:ip, :string, '127.0.0.1') { |ip| ... }
+  opts.on(:port, :int, default: 80) { |port| ... }
```
Comment (katzer) [inline README.md]: you forget to update the doc :)

Comment (HellRok) [reply README.md]: This has now been resolved, thank you for your patience!

**Review by katzer — APPROVED** [review]
LGTM

**Pull request merged**